wrapper.py

```
@@ -78,72 +79,102 @@ def main(logdir: str, cleanup: bool):
-    # This is a bit cumbersome but done as SLIMS doesn't have the run tag easily queryable right now
-    for sample in rnaseq_samples.keys():
-        # Get the runtag for the sample, combine with sample name
-        try:
-            slimsinfo = SlimsSample(sample)
-            runtag = find_runtag_from_fastqs(slimsinfo.fastqs)
-            sample_id = f"{sample}_{runtag}"
-        except Exception as e:
-            logger.error(e)
-            sys.exit(1)
-            # TODO, this should mark this sample as failed, and send an email. Not break the whole thing
-
-        if sample_id in previous_samples:
-            del rnaseq_samples[sample]
-
-    # Check if any samples remains, if so start the pipeline
+    # Find all DNA objects marked for QD-RNAseq pipeline as secondary analysis
+    # They also need to either be missing any bioinformatics objects set as QD-rnaseq,
+    # or have a bioinformatics object set as QD-rnaseq with the secAnalysisState set to 'novel'
+    rnaseq_samples = defaultdict(dict)
+    for sample in all_rnaseq_samples:
+        # Create the slimssample object
+        slimsinfo = SlimsSample(sample)
+        if not slimsinfo.fastqs:  # If no fastqs, skip
+            continue
+
+        for fastq in slimsinfo.fastqs:
+            # Skip fastq objects set to not include or no bioinformatics
+            if fastq.cntn_cstm_doNotInclude.value == True:
+                continue
+            elif fastq.cntn_cstm_noBioinformaticsObjects.value == True:
+                continue
+
+            # Get the runtag from the fastq object
+            runtag = fastq.cntn_cstm_runTag.value
+            sample_uniqID = "_".join([sample, runtag])
+
+            # Find all bioinformatics objects for this fastq object with correct secondary analysis
+            try:
+                attached_bioinfo_object = find_attached_bioinfo_objects(slimsinfo.bioinformatics, fastq.pk(), 186)
+            except Exception as e:
+                logger.error(e)
```

Should the code proceed on an exception raise? Might be missing a continue here or alike.

---

You're right. For now I've added a continue so the sample is skipped. What should also happen is that the sample should be marked as erroneous in SLIMS. I've added a todo comment for this for future PR